### ISSUE TITLE
Filter and ColReorder conflict

### ISSUE THREAD

**Issue by prubin73** [rstudio/DT#160]
In the following minimal example, filtering and then dragging columns around works correctly; but reordering and then filtering, or filtering, reordering and then adjusting/clearing the filter, results in the data reverting to the original column order but the column headings remaining in the manually altered arrangement (so mismatch between headers and data). The check box selects server or client side processing; both behave the same in the example. In the application in which I encountered this, behavior was more complex (including losing all data in the table if filtering, rearranging and then adjusting/clearing the filter with server-side processing but not with client-side processing).

``` r
library(shinydashboard)
library(DT)
shinyApp(
  ui = dashboardPage(
    dashboardHeader(title = "ColVis/Filter bug"),

    dashboardSidebar(
      checkboxInput("ss", "Server-side?", value = TRUE)
    ),

    dashboardBody(
      DT::dataTableOutput("data")
    )
  ),
  server =  function(input, output, session) {
    data <- USArrests

    output$data <- DT::renderDataTable(
      data,
      filter = "top",
      selection = "none",
      rownames = FALSE,
      extensions = c('ColReorder'),
      options = list(dom = 'C<"clear">Rlfrtip'),
      server = input$ss
    )
  }
)
```

**Comment by yihui** [comment]
Sorry, this one seems to be difficult to fix. I can confirm the issue, though.

```r
library(shiny)
library(shinydashboard)
library(DT)
shinyApp(
  ui = dashboardPage(
    dashboardHeader(title = "ColVis/Filter bug"),
    
    dashboardSidebar(
      checkboxInput("ss", "Server-side?", value = TRUE)
    ),
    
    dashboardBody(
      DT::dataTableOutput("data")
    )
  ),
  server =  function(input, output, session) {
    data <- USArrests
    
    output$data <- DT::renderDataTable(
      data,
      filter = "top",
      selection = "none",
      rownames = FALSE,
      extensions = c('ColReorder'),
      options = list(colReorder = TRUE),
      server = input$ss
    )
  }
)
```

**Comment by yihui** [comment]
This has been fixed by #532.
